app/controllers/v1/sessions_controller.rb

```diff
@@ -237,12 +252,16 @@ def originating_request_id
     end
 
     def url_service(previous_saml_uuid = nil)
-      SAML::URLService.new(saml_settings,
-                           session: @session_object,
-                           user: current_user,
-                           params: params,
-                           loa3_context: LOA::IDME_LOA3,
-                           previous_saml_uuid: previous_saml_uuid)
+      SAML::PostURLService.new(saml_settings,
```

Comment:
why not reuse SAML::URLService?  Whats different about PostURLService?

Reply:
Instead of returning a URL, it's returning the SSOe URL and a list of form parameters (but not jammed together into a query string),that are then used in this controller to render the auto-submitting form HTML template. 

But it's messy because _some_ of the interactions (such as SLO) are still a simple redirect vs. a render, so there's not really a clean interface to URLService/PostURLService.

Reply:
There's so much overlapping logic I'd like to bring those classes back together...thinking. about how to do that.